extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemLockupExtractor.java

```diff
@@ -0,0 +1,371 @@
+                .streamAsJsonObjects())
+            .map(thumbnailBadge -> thumbnailBadge.getObject("thumbnailBadgeViewModel"))
+            .anyMatch(thumbnailBadgeViewModel -> {
+                if ("THUMBNAIL_OVERLAY_BADGE_STYLE_LIVE".equals(
+                    thumbnailBadgeViewModel.getString("badgeStyle"))) {
+                    return true;
+                }
+
+                // Fallback: Check if there is a live icon
+                return thumbnailBadgeViewModel
+                    .getObject("icon")
+                    .getArray("sources")
+                    .streamAsJsonObjects()
+                    .map(source -> source
+                        .getObject("clientResource")
+                        .getString("imageName"))
+                    .anyMatch("LIVE"::equals);
+            })) {
+            return StreamType.LIVE_STREAM;
+        }
+
+        return StreamType.VIDEO_STREAM;
+    }
+
+    @Override
+    public boolean isAd() throws ParsingException {
+        final String name = getName(); // only get it once
+        return "[Private video]".equals(name)
+            || "[Deleted video]".equals(name);
+    }
+
+    @Override
+    public String getUrl() throws ParsingException {
+        try {
+            String videoId = lockupViewModel.getString("contentId");
+            if (isNullOrEmpty(videoId)) {
+                videoId = JsonUtils.getString(lockupViewModel,
+                    "rendererContext.commandContext.onTap.innertubeCommand.watchEndpoint.videoId");
+            }
+            return YoutubeStreamLinkHandlerFactory.getInstance().getUrl(videoId);
+        } catch (final Exception e) {
+            throw new ParsingException("Could not get url", e);
+        }
+    }
+
+    @Override
+    public String getName() throws ParsingException {
+        if (cachedName != null) {
+            return cachedName;
+        }
+
+        final String name = JsonUtils.getString(lockupViewModel,
+            "metadata.lockupMetadataViewModel.title.content");
+        if (!isNullOrEmpty(name)) {
+            this.cachedName = name;
+            return name;
+        }
+        throw new ParsingException("Could not get name");
+    }
+
+    @Override
+    public long getDuration() throws ParsingException {
+        final List<String> potentialDurations = JsonUtils.getArray(lockupViewModel,
+                "contentImage.thumbnailViewModel.overlays")
+            .streamAsJsonObjects()
+            .flatMap(jsonObject -> jsonObject
+                .getObject("thumbnailOverlayBadgeViewModel")
+                .getArray("thumbnailBadges")
+                .streamAsJsonObjects())
+            .map(jsonObject -> jsonObject
+                .getObject("thumbnailBadgeViewModel")
+                .getString("text"))
+            .collect(Collectors.toList());
+
+        if (potentialDurations.isEmpty()) {
+            throw new ParsingException("Could not get duration: No parsable durations detected");
+        }
```

Comment:
I think this has been not adjusted for lives.

Reply:
Done